src/lambda/config/templates.ts

```diff
@@ -250,19 +291,16 @@ export class TemplatesConfigPopulator {
         }
     }
 
-    private isUnexpectedObjectType(nodeType: jsonParser.NodeType): boolean {
-        return nodeType === 'array'
-            || nodeType === 'property'
-            || nodeType === 'string'
-            || nodeType === 'number'
-            || nodeType === 'boolean'
-    }
-
-    private ensureJsonObjectExists(jsonPath: jsonParser.JSONPath, value: any) {
+    private ensureJsonObjectExists(
+        jsonPath: jsonParser.JSONPath,
+        value: any,
+        allowedTypes: jsonParser.NodeType[] = ['object', 'null']
+    ) {
         const root = jsonParser.parseTree(this.json)
         const node = jsonParser.findNodeAtLocation(root, jsonPath)
+        const allowedTypesSet = new Set(allowedTypes)
 
-        if (node && this.isUnexpectedObjectType(node.type)) {
+        if (node && !allowedTypesSet.has(node.type)) {
```

Comment:
TemplatesConfigFieldTypeError should be adjusted to work with allowedTypesSet instead of 'expectedType: object'. This will also affect the showMessage and logging text in showTemplatesConfigurationError

Reply:
✔️